### PR TITLE
Correct Autoloading and STI guide (issue #26994)

### PR DESCRIPTION
**Pull request by kreintjes**
See the issue (#26994) for more information.

### CODE REVIEW

File: guides/source/autoloading_and_reloading_constants.md [hunk 1]
```diff
@@ -983,20 +983,19 @@ WHERE "polygons"."type" IN ("Rectangle")
...
-load the leaves of the tree by hand at the bottom of the file that defines the
-root class:
+manually load the direct subclasses at the bottom of the file that defines each
+intermediate class:
 
 ```ruby
-# app/models/polygon.rb
-class Polygon < ApplicationRecord
+# app/models/rectangle.rb
+class Rectangle < ApplicationRecord
```
Comment (matthewd) [inline guides/source/autoloading_and_reloading_constants.md]: This should inherit from `Polygon`, shouldn't it?

Comment (kreintjes) [inline guides/source/autoloading_and_reloading_constants.md]: You're right. Fixed :)

**Comment by kreintjes** [comment]
@matthewd @fxn what's the state of this? Anything I should still do?

**Comment by matthewd** [comment]
👌🏻

Thanks!

**Comment by fxn** [comment]
Why is this better? If you have A > B > C > D, with those lines in B and C, and you load C directly, you'll still get a circular dependency when C autoloads B?

Writing off the top of my head.

**Comment by fxn** [comment]
Why is this better? If you have A > B > C > D, with those lines in B and C, and you load C directly, you'll still get a circular dependency when C autoloads B?

Writing off the top of my head.

**Comment by matthewd** [comment]
@fxn maybe that's the case I was thinking of in https://github.com/rails/rails/issues/26994#issuecomment-260498757. I tried to come up with an example a few months ago, but couldn't work out how to break it.

**Comment by kreintjes** [comment]
@fxn @matthewd I have extended the demo application with a SubSquare extending from Square, see https://github.com/kreintjes/rails-autoloading-sti-demo/tree/sub-square. So now we have Polygon > Rectangle > Square > SubSquare. When I run `Square.all` in console on a cold boot, everything works fine:

```
Loading development environment (Rails 5.0.0.1)
irb(main):001:0> Square.all
  Square Load (0.1ms)  SELECT "polygons".* FROM "polygons" WHERE "polygons"."type" IN ('Square', 'SubSquare')
=> #<ActiveRecord::Relation [#<Square id: 1, type: "Square", created_at: "2017-05-09 09:30:50", updated_at: "2017-05-09 09:30:50">]>
```

So no circular dependency, not sure why though.

**Pull request merged**